.github/workflows/run_tests.yml

```diff
@@ -17,37 +17,39 @@ jobs:
        (github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'no_ci'))
     strategy:
       matrix:
-        os:
-          - name: ubuntu-latest
-          - name: windows-latest
-            msvc_bin: 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64'
-          - name: macos-latest
-            llvm: /opt/homebrew/opt/llvm
-          - name: macos-13
-            llvm: /usr/local/opt/llvm
+        os: [ubutu-latest, windows-latest, macos-latest, macos-13]
+
       fail-fast: false
-    runs-on: ${{ matrix.os.name }}
+
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
       - uses: conda-incubator/setup-miniconda@v3
         with:
           python-version: '3.10'
           channels: conda-forge,defaults
           channel-priority: true
+
+      - name: set HOMEBREW_PREFIX on Intel MacOS
+        if: matrix.os == 'macos-13'
+        run: |
+          echo HOMEBREW_PREFIX="/usr/local" >> $GITHUB_ENV
+
       - name: Install llvm on MacOS
-        if: startsWith(matrix.os.name, 'macos')
+        if: startsWith(matrix.os, 'macos')
         run: |
           brew install llvm
-          echo CC="${{ matrix.os.llvm }}/bin/clang" >> $GITHUB_ENV
-          echo LDFLAGS="-L${{ matrix.os.llvm }}/lib $LDFLAGS" >> $GITHUB_ENV
-          echo CPPFLAGS="-I${{ matrix.os.llvm }}/include $CPPFLAGS" >> $GITHUB_ENV
+          echo CC="${HOMEBREW_PREFIX}/opt/llvm/bin/clang" >> $GITHUB_ENV
+          echo LDFLAGS="-L${HOMEBREW_PREFIX}/opt/llvm/lib $LDFLAGS" >> $GITHUB_ENV
+          echo CPPFLAGS="-I${HOMEBREW_PREFIX}/opt/llvm/include $CPPFLAGS" >> $GITHUB_ENV
```

Comment:
Could use `env:` here to inject the constants for cleanliness?

Reply:
Is there a way to do matrux-dependent env? That would be ideal

Reply:
Looks like it is possible to put some logic in env: but it has to be done with && / || operators https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions

Reply:
How about just:
```sh
          LLVM_DIR=${HOMEBREW_PREFIX}/opt/llvm
          echo CC="${LLVM_DIR}/bin/clang" >> $GITHUB_ENV
          echo LDFLAGS="-L${LLVM_DIR}/lib $LDFLAGS" >> $GITHUB_ENV
          echo CPPFLAGS="-I${LLVM_DIR}/include $CPPFLAGS" >> $GITHUB_ENV
```

Reply:
I went for a blend of these 😄